packages/bpk-component-text/src/withTextStyle.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const withTextStyle = (WrappedComponent, type, tagName) => {
+  const component = (props) => {
+    const { children, ...rest } = props;
+
+    return (
+      <WrappedComponent
+        textStyle={type}
+        tagName={tagName}
+        {...rest}
```

Comment:
I think the HOC styles should win in this case - consumers will be forced to consider why thigns aren't working and realise that they probably shouldn't be using the HOC or vice versa.